tests/federation_room_join_partial_state_test.go

```diff
@@ -65,6 +65,17 @@ func TestPartialStateJoin(t *testing.T) {
 		return serverRoom
 	}
 
+	// getSyncToken gets the latest sync token
+	getSyncToken := func(t *testing.T, alice *client.CSAPI) string {
+		_, syncToken := alice.MustSync(t,
+			client.SyncReq{
+				Filter:        buildLazyLoadingSyncFilter(nil),
```

Comment:
I think you probably don't need a `Filter`?

Reply:
Won't that make the function block if we decide to use it midway through a partial join?

Reply:
yes, but I didn't think this was used in the middle of a partial-state join?

(indeed, I thought initialsyncs in the middle of partial-state joins blocked irrespective of the lazy-loading flag. I'm probably wrong though)

Reply:
It isn't currently used in the middle of a partial-state join, but I would like to keep the option open.

lazy-loading initial syncs don't (or no longer?) block during a partial-state join, otherwise one of the new lazy-loading tests in #442 would be getting stuck.

Reply:
all makes sense, thanks!